xbmc/XBDateTime.cpp

```diff
@@ -17,29 +17,17 @@
 
 #include <cstdlib>
 
-#define SECONDS_PER_DAY 86400L
-#define SECONDS_PER_HOUR 3600L
-#define SECONDS_PER_MINUTE 60L
-#define SECONDS_TO_FILETIME 10000000L
+#define USE_OS_TZDB 0
+#define HAS_REMOTE_API 0
+#include <date/date.h>
+#include <date/iso_week.h>
```

Comment:
date/iso_week.h is a private header. Isnt installed for 3.0.1 when using the tools/depends/target/date version
It is for 2.4.1, but as mentioned earlier, usage of c_encoding is then not possible.

Reply:
@lrusak added a step to install `iso_week.h`: https://github.com/xbmc/xbmc/blame/afd89125133bd0c025eed33a4479d18c932b15ee/cmake/modules/FindDate.cmake#L65

But I want to try Howard's proposed variant without private header file.

Reply:
Original commit was 2 years ago. A lot has probably changed since then 🙃

Reply:
Howard's proposed change for c_encoding doesn't remove the requirement of iso_week.h.
I've opened an issue on Howard's repo, as I think the additional headers including iso_week.h are meant to install but a cmake rework in 3.0.0 is no longer installing them. But after clarity from Howard first

Reply:
For now let's edit `Makefile` to copy `iso_week.h` and use 3.0.1 to avoid removing `c_encoding()`.

Reply:
We aren't on a rush here, so let's see what Howard responds to my issue. Who knows, maybe it is an oversight upstream with a potential fix to help the date project. Or otherwise changes in our code usage may need to be adapted

Reply:
I noticed the same issue when building for windows so I tweaked the CMakeLists.txt to include private headers as well https://github.com/Paxxi/date/commit/e9943526c2ba23746b3fe4fb0522be85b1efcd00

Reply:
I noticed this issue a year ago and added the patch to expose private headers. I did not publish it though.

Reply:
I haven't followed this discussion in detail.  But `iso_week.h` is a stand-alone, header-only library, documented here: https://howardhinnant.github.io/date/iso_week.html

Its purpose is two-fold:

1.  Create a calendrical system that can be specified in terms of the ISO week-based year, for example: `sat/51/2015`.
2. Demonstrate that user-written calendrical systems, in their own namespace, can interoperate with the facilities in the `date` namespace.

Whether or not a CMake file includes `iso_week.h` is not a decision I ever made, as I don't personally maintain (or recommend) the CMake files at my repository.  In hindsight it was a mistake for me to ever OK their placement on my site.

Reply:
@HowardHinnant but since it is there, do you accept exposing these headers to the broader public? I think it makes no harm to expose them. Similar situation was in kissfft where I overhauled both CMake and Make scripts.

Reply:
I think it is fine to expose them to the public, else I wouldn't have posted and documented them. :-)  I'm happy if people find them useful.  `ptz.h` has probably been the most useful, and yet I've failed to document it.  It demonstrates that a user can write a custom time zone and use it with `date::zoned_time`.  It implements `Posix::time_zone`.  Here is an example use:
```c++
#include "date/ptz.h"
#include <iostream>

int
main()
{
    using namespace date;
    using namespace std;
    using namespace std::chrono;

    auto now = system_clock::now();
    auto tzp = Posix::time_zone{"EST5EDT,M3.2.0,M11.1.0"};
    cout << zoned_time{tzp, now} << '\n';
}
``` 
This just print out for me:
```
2021-11-14 12:51:28.079505 EST
```
This is a way to use POSIX time zones instead of IANA time zones, and `ptz.h` is header-only.  No need to deal with compiling `tz.cpp` or any of the other installation issues associated with `tz.h`.